File: Leet Code/Python/problem_29.py
class Solution(object):
    def divide(self, dividend: int, divisor: int) -> int:
        """
        :type dividend: int
        :type divisor: int
        :rtype: int
        """ 
        max = pow(2, 31) - 1
        min = pow(-2, 31)
        if (dividend < 0 and divisor < 0 or dividend > 0 and divisor > 0) :
            quotient = (int)(dividend / divisor)
            if ( quotient > max) :
                return max
            return quotient
        
        if (dividend < 0 or divisor < 0) :
            quotient = (int)(dividend / divisor)
            if (quotient < min) :
                return min
            return quotient
        
        return 0


sol = Solution()
result = sol.divide(7, -3)
print(result)

File: Leet Code/JavaScript/problem_29.js
const divide = function(dividend, divisor) {
    let max = Math.pow(2, 31) - 1;
    let min = Math.pow(-2, 31);
    if (dividend < 0 && divisor < 0 || dividend > 0 && divisor > 0) {
        let quotient = Math.trunc(dividend / divisor);
        if (quotient > max) {
            return max;
        };
        return quotient;
    };

    if (dividend < 0 || divisor < 0) {
        let quotient = Math.trunc(dividend / divisor);
        if (quotient < min) {
            return min;
        };
        return quotient;
    };
    return 0;
};

const result = divide(7, -3);
console.log(result);